### PR TITLE
add autoscaler for queriers

### PR DESCRIPTION
**Pull request by kd7lxl**
**What this PR does**:
Adds a `HorizontalPodAutoscaler` for the queriers.

https://cortexmetrics.io/docs/architecture/#querier
> Queriers are stateless and can be scaled up and down as needed.

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -819,6 +819,17 @@ querier:
...
 
   annotations: {}
+
+  autoscaling:
+    # -- Creates a HorizontalPodAutoscaler for the querier pods.
+    enabled: false
+    minReplicas: 2
+    maxReplicas: 30
+    targetCPUUtilizationPercentage: 80
+    targetMemoryUtilizationPercentage: 0  # 80
```
Comment (ShuzZzle) [inline values.yaml:829]: nice but i don't like the defaults. querier is also a highly dependant memory process, even though query frontend already splits querier by day.

I actually don't think CPU matters that much

**Review by ShuzZzle — CHANGES_REQUESTED** [review]

**Review by ShuzZzle — APPROVED** [review]

File: values.yaml (hunk 1 above)
Comment (ShuzZzle) [reply values.yaml:829]: @cabrinha what's your opinion?

Comment (kd7lxl) [reply values.yaml:829]: Somewhat related, I've spent a lot of time tuning resource definitions for each Cortex component and still don't feel like I've got it dialed in. I'd appreciate any real world knowledge people could share in the wiki or elsewhere.

Comment (ShuzZzle) [reply values.yaml:829]: >  still don't feel like I've got it dialed in

Yep its like a never ending story :)

**Comment by cabrinha** [comment]
It'd also be nice if we could add HPA for every component that can safely be autoscaled.

**Comment by kd7lxl** [comment]
> It'd also be nice if we could add HPA for every component that can safely be autoscaled.

I've been adding them one at a time as a learn/recognize that they are safe to autoscale, and by what metric. If someone with more operational experience could share appropriate scaling metrics for each component, I could add the autoscalers.

**Pull request merged**